src/index.ts

```diff
@@ -988,52 +1050,35 @@ export class SupraClient {
         "transfer",
         [],
         [receiverAccountAddr.toUint8Array(), BCS.bcsSerializeUint64(amount)],
-        this.chainId,
-        maxGas
+        optionalTransactionArgs?.optionalTransactionPayloadArgs
       )
     );
 
     return await this.sendTx(
       sendTxPayload,
-      enableSimulation,
-      waitForTransactionCompletion
+      optionalTransactionArgs?.enableTransactionWaitAndSimulationArgs
     );
   }
 
   /**
-   * Transfer coin
+   * Transfer custom type of coin
    * @param senderAccount Sender KeyPair
    * @param receiverAccountAddr Receiver Supra Account address
    * @param amount Amount to transfer
-   * @param coinType Type of coin
-   * @param enableSimulation should enable simulation
-   * @param waitForTransactionCompletion should wait for transaction completion
+   * @param coinType Type of custom coin
+   * @param optionalTransactionArgs optional arguments for transaction
    * @returns `TransactionResponse`
    */
   async transferCoin(
     senderAccount: SupraAccount,
     receiverAccountAddr: HexString,
     amount: bigint,
     coinType: string,
-    enableSimulation: boolean = DEFAULT_ENABLE_SIMULATION,
-    waitForTransactionCompletion: boolean = DEFAULT_WAIT_FOR_TX_COMPLETION
+    optionalTransactionArgs?: OptionalTransactionArgs
   ): Promise<TransactionResponse> {
-    let maxGas = BigInt(50000);
-    if (
-      BigInt(0) + maxGas * BigInt(100) >
-      (await this.getAccountSupraCoinBalance(senderAccount.address()))
-    ) {
-      throw new Error("Insufficient Supra Coins");
-    }
-    if (
-      BigInt(amount) >
-      (await this.getAccountCoinBalance(senderAccount.address(), coinType))
```

Comment:
Why have these checks been removed?

Reply:
This was a redundant check, as we are doing simulation before tx